workers/unity/Assets/Fps/Scripts/Config/FpsEntityTemplates.cs

```diff
@@ -51,43 +57,52 @@ public static EntityTemplate Player(string workerId, Vector3f position)
-            var maxHealth = PlayerHealthSettings.MaxHealth;
-
-            var healthComponent = HealthComponent.Component.CreateSchemaComponentData(maxHealth, maxHealth);
-            var healthRegenComponent = HealthRegenComponent.Component.CreateSchemaComponentData(false,
-                0,
-                PlayerHealthSettings.SpatialCooldownSyncInterval,
-                PlayerHealthSettings.RegenAfterDamageCooldown,
-                PlayerHealthSettings.RegenInterval,
-                PlayerHealthSettings.RegenAmount);
-
-            return EntityBuilder.Begin()
-                .AddPosition(spawnPosition.x, spawnPosition.y, spawnPosition.z, gameLogic)
-                .AddMetadata("Player", gameLogic)
-                .SetPersistence(false)
-                .SetReadAcl(AllWorkerAttributes)
-                .AddComponent(serverMovement, gameLogic)
-                .AddComponent(clientMovement, client)
-                .AddComponent(clientRotation, client)
-                .AddComponent(shootingComponent, client)
-                .AddComponent(gunComponent, gameLogic)
-                .AddComponent(gunStateComponent, client)
-                .AddComponent(healthComponent, gameLogic)
-                .AddComponent(healthRegenComponent, gameLogic)
-                .AddPlayerLifecycleComponents(workerId, client, gameLogic)
-                .Build();
+
+            var pos = new Position.Snapshot { Coords = spawnPosition.ToSpatialCoordinates() };
+            var serverMovement = new ServerMovement.Snapshot { Latest = serverResponse };
+            var clientMovement = new ClientMovement.Snapshot { Latest = new ClientRequest() };
+            var clientRotation = new ClientRotation.Snapshot { Latest = rotationUpdate };
+            var shootingComponent = new ShootingComponent.Snapshot();
+            var gunComponent = new GunComponent.Snapshot { GunId = PlayerGunSettings.DefaultGunIndex };
+            var gunStateComponent = new GunStateComponent.Snapshot { IsAiming = false };
+            var healthComponent = new HealthComponent.Snapshot
+            {
+                Health = PlayerHealthSettings.MaxHealth,
+                MaxHealth = PlayerHealthSettings.MaxHealth,
+            };
+
+            var healthRegenComponent = new HealthRegenComponent.Snapshot
+            {
+                CooldownSyncInterval = PlayerHealthSettings.SpatialCooldownSyncInterval,
+                DamagedRecently = false,
+                RegenAmount = PlayerHealthSettings.RegenAmount,
+                RegenCooldownTimer = PlayerHealthSettings.RegenAfterDamageCooldown,
+                RegenInterval = PlayerHealthSettings.RegenInterval,
+                RegenPauseTime = 0,
+            };
+
+            var template = new EntityTemplate();
+            template.AddComponent(pos, WorkerUtils.UnityGameLogic);
+            template.AddComponent(new Metadata.Snapshot { EntityType = "Player" }, WorkerUtils.UnityGameLogic);
```

Comment:
Any reason this is the only one where we inline the instantiation?

Reply:
cause most of the others are longer, so I split them into multiple lines to make it easier to read